internal/test/framework/windowsvm.go

```diff
@@ -181,53 +181,58 @@ func (w *windowsVM) RetrieveFiles(remoteDir, localDir string) error {
-		// in a directory, if this is directory, we should have called RetrieveFiles on this directory
+		remotePath := filepath.Join(remoteDir, remoteFile.Name())
+		localPath := filepath.Join(localDir, remoteFile.Name())
+		// check if it is a directory, call itself again
 		if remoteFile.IsDir() {
-			continue
-		}
-		fileName := remoteFile.Name()
-		dstFile, err := os.Create(filepath.Join(localDir, fileName))
-		if err != nil {
-			log.Printf("error creating file locally: %v", err)
-			continue
-		}
-		// TODO: Check if there is some performance implication of multiple Open calls.
-		srcFile, err := sftp.Open(remoteDir + "\\" + fileName)
-
-		if err != nil {
-			log.Printf("error while opening remote directory on the Windows VM: %v", err)
-			continue
-		}
-		_, err = io.Copy(dstFile, srcFile)
-		if err != nil {
-			log.Printf("error retrieving file %v from Windows VM: %v", fileName, err)
-			continue
-		}
-		// flush memory
-		if err = dstFile.Sync(); err != nil {
-			log.Printf("error flusing memory: %v", err)
-			continue
-		}
-		if err := srcFile.Close(); err != nil {
-			log.Printf("error closing file on the remote host %s", fileName)
-			continue
-		}
-		if err := dstFile.Close(); err != nil {
-			log.Printf("error closing file %s locally", fileName)
-			continue
+			if err = w.RetrieveDirectories(remotePath, localPath); err != nil {
+				log.Printf("error while retrieving directiories from Windows VM %s : %v", remotePath, err)
+			}
+		} else {
+			// best effort to retrieve files
+			w.retrieveFiles(sftp, remotePath, localPath)
 		}
 	}
 	return nil
 }
 
+// retrieveFiles retrieves list of files from remote directory to the local directory.
+// we're doing a best effort to collect every log possible. If a retrieval of file fails,
+// we would proceed with retrieval of other log files.
+func (w *windowsVM) retrieveFiles(sftp *sftp.Client, remotePath, localPath string) {
+
+	localFile, err := os.Create(localPath)
+	if err != nil {
+		log.Printf("error creating file locally: %v", err)
+	}
+	// TODO: Check if there is some performance implication of multiple Open calls.
+	remoteFile, err := sftp.Open(remotePath)
+	if err != nil {
+		log.Printf("error while opening remote file on the Windows VM: %v", err)
+	}
+	_, err = io.Copy(localFile, remoteFile)
+	if err != nil {
+		log.Printf("error retrieving file %v from Windows VM: %v", localPath, err)
+	}
+	// flush memory
+	if err = localFile.Sync(); err != nil {
+		log.Printf("error flusing memory: %v", err)
+	}
+	if err := remoteFile.Close(); err != nil {
+		log.Printf("error closing file on the remote host %s", localPath)
+	}
+	if err := localFile.Close(); err != nil {
+		log.Printf("error closing file %s locally", localPath)
+	}
+}
+
 func (w *windowsVM) Run(cmd string, psCmd bool) (string, string, error) {
 	if w.winrmClient == nil {
-		return "", "", fmt.Errorf("Run cannot be called without a WinRM client")
+		return "", "", fmt.Errorf("run cannot be called without a WinRM client")
```

Comment:
I would also love for this to be changed, but its unrelated to the PR
Its good that you caught this, but we want to avoid doing random fixes within tickets. :)
But maybe its ok in this case? I'll let others chime in here